packages/arb-token-bridge-ui/src/components/common/SearchPanel/SearchPanel.tsx

```diff
@@ -1,128 +1,155 @@
-import { PropsWithChildren, useCallback, useState } from 'react'
-import { ArrowSmallLeftIcon, XMarkIcon } from '@heroicons/react/24/outline'
+import React, {
+  PropsWithChildren,
+  createContext,
+  useCallback,
+  useContext,
+  useMemo,
+  useState
+} from 'react'
+import {
+  ArrowRightIcon,
+  ArrowSmallLeftIcon,
+  XMarkIcon
+} from '@heroicons/react/24/outline'
+
 import { Loader } from '../atoms/Loader'
 
 enum Panel {
   MAIN,
   SECONDARY
 }
 
+function usePanel(panel: Panel) {
+  const { currentPanel, setCurrentPanel } = useContext(SearchPanelContext)
+  const setPanel = useCallback(
+    () => setCurrentPanel(panel),
+    [panel, setCurrentPanel]
+  )
+  const isActivePanel = currentPanel === panel
+
+  return {
+    setPanel,
+    isActivePanel
+  }
+}
+
+const SearchPanelContext = createContext<{
+  currentPanel: Panel
+  setCurrentPanel: (panel: Panel) => void
+  // eslint-disable-next-line @typescript-eslint/no-empty-function
+}>({ currentPanel: Panel.MAIN, setCurrentPanel: () => {} })
+
+function SearchPanel({ children }: { children: React.ReactNode }) {
+  const [currentPanel, setCurrentPanel] = useState(Panel.MAIN)
+  const provided = useMemo(
+    () => ({
+      currentPanel,
+      setCurrentPanel: (panel: Panel) => setCurrentPanel(panel)
+    }),
+    [currentPanel]
+  )
+  return (
+    <SearchPanelContext.Provider value={provided}>
+      {children}
+    </SearchPanelContext.Provider>
+  )
+}
+
+function MainPageCTA({ children }: { children: React.ReactNode }) {
+  const { setPanel: showSecondaryPage } = usePanel(Panel.SECONDARY)
+  return (
+    <button
+      className="arb-hover text-gray flex items-center gap-2 text-sm"
+      onClick={showSecondaryPage}
+    >
+      <span>{children}</span>
+      <ArrowRightIcon className="h-3 w-3 stroke-[3px]" />
+    </button>
+  )
+}
+SearchPanel.MainPageCTA = MainPageCTA
+
+function CloseButton({ onClick }: React.HTMLAttributes<HTMLButtonElement>) {
```

Comment:
We may as well allow all props to be passed for more customization in the future?